core/remotes/docker/resolver.go

```diff
@@ -374,6 +374,12 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 					return "", ocispec.Descriptor{}, err
 				}
 
+				// Check for error status code
+				if resp.StatusCode >= http.StatusBadRequest && resp.StatusCode < http.StatusNetworkAuthenticationRequired {
```

Comment:
Why is `StatusNetworkAuthenticationRequired` excluded? Does this require handling differently from other errors?

Reply:
https://github.com/containerd/containerd/pull/12262/commits/84998ad0409afded12bbe090cfe93de1bfb41b17

Reply:
https://github.com/containerd/containerd/pull/12262/commits/d7054f63bca36a8b11858bcf18c4021c68db3ad2